camel/toolkits/arxiv_toolkit.py

```diff
@@ -0,0 +1,139 @@
+
+
+class ArxivToolkit(BaseToolkit):
+    r"""A toolkit for interacting with the arXiv API to search and download
+    academic papers.
+    """
+
+    @dependencies_required('arxiv')
+    def __init__(self) -> None:
+        r"""Initializes the ArxivToolkit and sets up the arXiv client."""
+        import arxiv
+
+        self.client = arxiv.Client()
+
+    def _get_search_results(
+        self,
+        query: str,
+        paper_ids: Optional[List[str]] = None,
+        max_results: Optional[int] = 5,
+    ) -> Generator:
+        r"""Retrieves search results from the arXiv API based on the provided
+        query and optional paper IDs.
+
+        Args:
+            query (str): The search query string used to search for papers on
+                arXiv.
+            paper_ids (List[str], optional): A list of specific arXiv paper
+                IDs to search for. (default::obj: `None`)
+            max_results (int, optional): The maximum number of search results
+                to retrieve. (default::obj: `5`)
+
+        Returns:
+            Generator: A generator that yields results from the arXiv search
+                query, which includes metadata about each paper matching the
+                query.
+        """
+        import arxiv
+
+        paper_ids = paper_ids or []
+        search_query = arxiv.Search(
+            query=query,
+            id_list=paper_ids,
+            max_results=max_results,
+        )
+        return self.client.results(search_query)
+
+    def search_papers(
+        self,
+        query: str,
+        paper_ids: Optional[List[str]] = None,
+        max_results: Optional[int] = 5,
+    ) -> List[Dict[str, str]]:
+        r"""Searches for academic papers on arXiv using a query string and
+        optional paper IDs.
+
+        Args:
+            query (str): The search query string.
+            paper_ids (List[str], optional): A list of specific arXiv paper
+                IDs to search for. (default::obj: `None`)
+            max_results (int, optional): The maximum number of search results
+                to return. (default::obj: `5`)
+
+        Returns:
+            List[Dict[str, str]]: A list of dictionaries, each containing
+                information about a paper,
+            including title, published date, authors, entry ID, summary, and
+                extracted text from the paper.
+        """
+        from arxiv2text import arxiv_to_text
+
+        search_results = self._get_search_results(
+            query, paper_ids, max_results
+        )
+        papers_data = []
+
+        for paper in search_results:
+            paper_info = {
+                "title": paper.title,
+                "published_date": paper.updated.date().isoformat(),
+                "authors": [author.name for author in paper.authors],
+                "entry_id": paper.entry_id,
+                "summary": paper.summary,
+                "paper_text": arxiv_to_text(paper.pdf_url),
```

Comment:
Should we use `unstructured io` instead for better performance?

Reply:
Hey @lightaime , after conducting a comparison test between `unstructured io` and `arxiv_to_text`, I found that `unstructured io` struggles with papers that have content formatted in two columns on a single page, as it ends up scrambling the text. However, `arxiv_to_text` handles this type of formatting well.

<img width="812" alt="Screenshot 2024-10-08 at 14 25 25" src="https://github.com/user-attachments/assets/5c7d4d56-f5df-4d4c-bdcf-11e237ddee9d">


<img width="998" alt="Screenshot 2024-10-08 at 14 28 22" src="https://github.com/user-attachments/assets/f77a44dc-8bd3-4740-b0e9-18f6217164c3">